### PR TITLE
Refactor dyn adjust panel with tabs, implement the importance adjustment

### PR DESCRIPTION
**Pull request by echozzy629**

### CODE REVIEW

File: src/composites/ConfigurationContainer/ImportanceAdjustment.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+export const ImportanceAdjustment = () => {
+    const [selectedProfile, setSelectedProfile] = useState<Profile | Profile[] | null>(null);
+
+    // haddle the profile selection and apply
+    const [isProfileApplied, setIsProfileApplied] = useState(false);
+
+    const handleProfileApply = (profile: Profile[] | null) => {
+        setSelectedProfile(profile);
+        setIsProfileApplied(true);
```
Comment (devjeetr) [inline src/composites/ConfigurationContainer/ImportanceAdjustment.tsx]: You can use `const isProfileApplied = profile === null` to achieve the same effect. When you want to reset the profile, set the selected profile to `null`. This will avoid issues of `selectedProfile` and `isProfileApplied` going out of sync.

File: src/composites/ConfigurationContainer/ImportanceAdjustment/AdjsutmentTable.tsx [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+    const [sliderValues, setSliderValues] = useState<Weights>(getInitialWeights);
+
+    // Reset sliders to initial weights (dataset or selected profile)
+    const resetAllAdjustments = () => {
+        setSliderValues(getInitialWeights(true)); // Force use of dataset weights
+        onResetApplied(); // Signal to parent that reset was triggered
+    };
+
+   // Calculate total importance for normalization
+   const totalImportance = useMemo(() => Object.values(sliderValues).reduce((sum, importance) => sum + importance, 0), [sliderValues]);
```
Comment (devjeetr) [inline src/composites/ConfigurationContainer/ImportanceAdjustment/AdjsutmentTable.tsx]: good stuff :)

File: src/composites/ConfigurationContainer/ImportanceAdjustment/AdjsutmentTable.tsx [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+    // Handle slider value change
+    const handleSliderChange = (name: string, newImportance: number) => {
+        setSliderValues(prev => ({ ...prev, [name]: newImportance }));
+    };
+
+    // Update sliderValues when selectedProfile changes
+    if (isProfileApplied) {
+        const newInitialWeights = getInitialWeights();
+        if (JSON.stringify(newInitialWeights) !== JSON.stringify(sliderValues)) {
+            setSliderValues(newInitialWeights);
```
Comment (devjeetr) [inline src/composites/ConfigurationContainer/ImportanceAdjustment/AdjsutmentTable.tsx]: hmm this will cause problems, since `isProfileApplied` comes from props and this state change relies on changes to props. This can run into an infinite loop. One solution is to pass slider values from the parent directly. Is there any reason why these cannot be derived directly in a `useMemo`?

File: src/composites/ConfigurationContainer/ImportanceAdjustment/ProfileSelection.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+export const ProfileSelection: React.FC<ProfileSelectionProps> = ({ onProfileChange }) => {
+    // Use state to manage the dynamically uploaded or customized profiles
+    const [selectedProfileType, setSelectedProfileType] = useState<string | null>(null);
+    const [uploadedProfiles, setUploadedProfiles] = useState<Profile[]>([]);
+
+    // Function to handle profile selection changes
+    const handleProfileChange = (value: string) => {
+        setSelectedProfileType(value);
```
Comment (devjeetr) [inline src/composites/ConfigurationContainer/ImportanceAdjustment/ProfileSelection.tsx]: typically, we don't want to maintain state locally(`setSelectedProfileType`) but also in the parent component (`onProfileChange`). You probably want to use a state on the parent component for this info, and this component should just receive those props, and alert the parent on user interactions.

**Review by devjeetr — CHANGES_REQUESTED** [review]

File: src/composites/ConfigurationContainer/ImportanceAdjustment.tsx (hunk 1 above)
Comment (echozzy629) [reply src/composites/ConfigurationContainer/ImportanceAdjustment.tsx]: addressed

File: src/composites/ConfigurationContainer/ImportanceAdjustment/AdjsutmentTable.tsx (hunk 3 above)
Comment (echozzy629) [reply src/composites/ConfigurationContainer/ImportanceAdjustment/AdjsutmentTable.tsx]: updated with useMemo

File: src/composites/ConfigurationContainer/ImportanceAdjustment/ProfileSelection.tsx (hunk 4 above)
Comment (echozzy629) [reply src/composites/ConfigurationContainer/ImportanceAdjustment/ProfileSelection.tsx]: addressed, rn state setSelectedProfile is set in the parent component, and the selectedProfile is passed to the child component (ProfileSelection)

File: src/composites/ConfigurationContainer/ImportanceAdjustment/AdjsutmentTable.tsx [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+
+  // useMemo to calculate initial weights based on selectedProfile or dataset
+  const sliderValues = useMemo(() => {
+    const useDataset = !isProfileApplied;
+    return getInitialWeights(selectedProfile, dataset, useDataset);
+  }, [selectedProfile, dataset, isProfileApplied]);
+
+  const [values, setValues] = useState<{ [key: string]: number }>(
+    () => sliderValues
+  );
```
Comment (devjeetr) [inline src/composites/ConfigurationContainer/ImportanceAdjustment/AdjsutmentTable.tsx]: this will go stale i.e. `values` is only ever equal to `sliderValues` the first time you render. If you recalculate sliderValues, `values` won't update. 

I think we still have the same problem as before. There needs to be a single prop representing slider values instead of several, and from the usage of `onResetApplied` it appears that that state shouldn't be in this component. 

What exactly is the overall chain of operations you're trying to model here?

File: src/composites/ConfigurationContainer/ImportanceAdjustment.tsx [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+              {/*Profile selection*/}
+              <ProfileSelection
+                onProfileChange={handleProfileApply}
+                selectedProfile={selectedProfile}
+              />
+
+              <Separator my="3" size="4" />
+
+              {/* show the adjustment table*/}
+              <AdjustmentTable
```
Comment (devjeetr) [inline src/composites/ConfigurationContainer/ImportanceAdjustment.tsx:117]: you should be holding the slider values in this component, and passing them into `AdjustmentTable` here.

File: src/composites/ConfigurationContainer/ImportanceAdjustment/AdjsutmentTable.tsx
Comment (devjeetr) [inline src/composites/ConfigurationContainer/ImportanceAdjustment/AdjsutmentTable.tsx:1]: filename

File: src/data/schema.ts [hunk 7]
```diff
@@ -16,6 +16,7 @@ const utilityFunctionSchema = z.union([z.string(), utilityFunctionNewStructure])
 
 export namespace base {
   export const measureSingle = z.object({
+    name: z.string(),
```
Comment (devjeetr) [inline src/data/schema.ts]: next time this should be in a separate PR, because otherwise this won't get merged until the rest of the PR gets merged

**Review by devjeetr — CHANGES_REQUESTED** [review]

File: src/data/schema.ts (hunk 7 above)
Comment (echozzy629) [reply src/data/schema.ts]: Gotcha! I forgot i have already addressed this in the refactor branch... Next time will make another PR for this type of changes.

File: src/composites/ConfigurationContainer/ImportanceAdjustment/AdjsutmentTable.tsx
Comment (echozzy629) [reply src/composites/ConfigurationContainer/ImportanceAdjustment/AdjsutmentTable.tsx:1]: fixed this typo, also updated corresponding imports

File: src/composites/ConfigurationContainer/ImportanceAdjustment/AdjsutmentTable.tsx (hunk 5 above)
Comment (echozzy629) [reply src/composites/ConfigurationContainer/ImportanceAdjustment/AdjsutmentTable.tsx]: the overall flow should be: by default in the configuration "adjust importance" tab, the user could decide whether to use a pre-defined profile or not (`ProfileSelection` should return either the selected profile or the default weights in the initial dataset.), based on the selected profile, `AdjustmentTable` will show sliders whose default values as the profile, and users are able to adjust each slider.

Do you have any suggestions? My previous idea is to get selected profile from `ProfileSelection`,  and `ImportanceAdjustment` passes it `AdjustmentTable` to set it as the initial slider default value. A local state `values` in `AdjustmentTable` is for tracking the sliders adjustment, which is based on the selected profile.

File: src/composites/ConfigurationContainer/ImportanceAdjustment.tsx (hunk 6 above)
Comment (echozzy629) [reply src/composites/ConfigurationContainer/ImportanceAdjustment.tsx:117]: standby for rn, will fix this together with the previous comment.

File: src/composites/ConfigurationContainer/ImportanceAdjustment/AdjsutmentTable.tsx (hunk 5 above)
Comment (devjeetr) [reply src/composites/ConfigurationContainer/ImportanceAdjustment/AdjsutmentTable.tsx]: pending offline discussion

**Review by devjeetr — APPROVED** [review]
